pocs/camera/camera.py

```diff
@@ -496,14 +480,44 @@ def _process_fits(self, file_path, info):
         fits_utils.update_headers(file_path, info)
         return file_path
 
+    def _create_subcomponent(self, subcomponent, name, class_name, base_class):
+        """
+
+        """
+        if subcomponent:
+            if isinstance(subcomponent, base_class):
+                self.logger.debug("{} received: {}".format(class_name, subcomponent))
+                setattr(self, name, subcomponent)
+                getattr(self, name).camera = self
+            elif isinstance(subcomponent, dict):
+                module_name = 'pocs.{}.{}'.format(name, subcomponent['model'])
+                try:
+                    module = load_module(module_name)
+                except AttributeError as err:
+                    self.logger.critical("Couldn't import {} module {}!".format(
+                        class_name, module_name))
+                    raise err
+                else:
+                    subcomponent_kwargs = copy.copy(subcomponent)
+                    subcomponent_kwargs.update({'camera': self, 'config': self.config})
+                    setattr(self, name, getattr(module, class_name)(**subcomponent_kwargs))
+            else:
+                # Should have been passed either an instance of base_class or dict with subcomponent
+                # configuration. Got something else...
+                self.logger.error("Expected either a {} instance or dict, got {}".format(
+                    class_name, subcomponent))
+                setattr(self, name, None)
+        else:
+            setattr(self, name, None)
+
     def __str__(self):
         name = self.name
         if self.is_primary:
             name += ' [Primary]'
 
         s = "{} ({}) on {}".format(name, self.uid, self.port)
```

Comment:
With the default `name=None` this string might look weird.

Reply:
In `AbstractCamera` the default for the `name` attribute is 'Generic Camera', and the SBIG and FLI `Camera` classes replace that with their own defaults ('SBIG Camera' and 'FLI Camera').  Might end up looking a bit odd with Canon cameras though, which don't change the default from 'Generic Camera'.

Reply:
D'oh, yeah, sorry, I had a brain lapse and was thinking the `setattr` from above would overwrite that, but they are using the variable `name`, not the "name".

Reply:
Ah, yes. I should probably make the variable name in the method about something more specific than `name` to avoid that sort of confusion.

Reply:
Renamed to `name` to `sub_name` in order to be a bit less ambiguous.